src/coreclr/jit/lsra.cpp

```diff
@@ -733,16 +735,15 @@ LinearScan::LinearScan(Compiler* theCompiler)
             availableRegs[i] = &availableDoubleRegs;
         }
 #ifdef FEATURE_SIMD
-#if defined(TARGET_XARCH)
-        else if ((thisType >= TYP_SIMD8) && (thisType <= TYP_SIMD32))
-#else
-        else if ((thisType >= TYP_SIMD8) && (thisType <= TYP_SIMD16))
-#endif // TARGET_XARCH
-
+        else if (varTypeIsSIMD(thisType))
         {
             availableRegs[i] = &availableDoubleRegs;
         }
-#endif
+        else if (thisType == TYP_MASK)
```

Comment:
This should be under `#ifdef TARGET_AMD64`. That's why there are seeing arm64 TP regressions. Also, use `varTypeIsMask()`.

Reply:
I think we want `#ifdef TARGET_XARCH` but yes, I will make the change.